lib/serve-asset-middleware.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+var serveAsset = require('./utils/serve-asset');
+
+module.exports = function serveAssetsMiddleware() {
+  return function(request, response, next) {
+    var broccoliInfo = request.headers['x-broccoli'];
+
+    if (broccoliInfo && broccoliInfo.filename) {
+      serveAsset(response, {
+        'filename': broccoliInfo.filename
+      });
+    } else {
+      // bypass this middleware if the filename is not defined
+      next();
```

Comment:
In `ember-cli` the last middleware in the chain is the [proxy-server-middleware](https://github.com/ember-cli/ember-cli/blob/master/lib/tasks/server/middleware/proxy-server/index.js) which is invoked when a file is not found in disk or if index.html and autoIndex is not true. Basically the case when someone uses `ember serve --proxy http://localhost:3000` for example. In this case, when we know the file is not found we want to bypass this middleware (ie not even serve it) and let the proxy server handle the request.